### PR TITLE
fix hack/make-rules/make-help.sh shellcheck failures

### PR DESCRIPTION
**Pull request by BenTheElder**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug

/kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**: fixes shellcheck failures in `hack/make-rules/make-help.sh`

**Which issue(s) this PR fixes**:

Part of https://github.com/kubernetes/kubernetes/issues/72956

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

/sig release
/priority important-longterm

### CODE REVIEW

File: hack/make-rules/make-help.sh [hunk 1]
```diff
@@ -21,22 +21,21 @@ set -o pipefail
 readonly red=$(tput setaf 1)
 readonly reset=$(tput sgr0)
 
-KUBE_ROOT=$(dirname "${BASH_SOURCE}")/../..
+KUBE_ROOT=$(dirname "${BASH_SOURCE[0]}")/../..
 ALL_TARGETS=$(make -C "${KUBE_ROOT}" PRINT_HELP=y -rpn | sed -n -e '/^$/ { n ; /^[^ .#][^ ]*:/ { s/:.*$// ; p ; } ; }' | sort)
-CMD_TARGETS=$(ls -l "${KUBE_ROOT}/cmd" |awk '/^d/ {print $NF}')
+CMD_TARGETS=$(cd "${KUBE_ROOT}/cmd"; find . -type d -mindepth 1 -maxdepth 1 | cut -c 3-)
```
Comment (BenTheElder) [inline hack/make-rules/make-help.sh]: ls output is not reliably portable / not specified. this version should be more portable and doesn't require knowing awk 😉 

same as https://github.com/kubernetes/kubernetes/pull/79316 we're `cd`ing to do find relative to `.` and using `cut -c 3-` to strip the leading `./`

Comment (fejta) [reply hack/make-rules/make-help.sh]: `find $KUBEROOT/cmd -type d`

Comment (BenTheElder) [reply hack/make-rules/make-help.sh]: if we do that we have to strip the leading `$KUBEROOT` which may be some arbitrary absolute path, which is more complex than stripping the leading `./`.

File: hack/make-rules/make-help.sh [hunk 2]
```diff
@@ -21,22 +21,21 @@ set -o pipefail
 readonly red=$(tput setaf 1)
 readonly reset=$(tput sgr0)
 
-KUBE_ROOT=$(dirname "${BASH_SOURCE}")/../..
+KUBE_ROOT=$(dirname "${BASH_SOURCE[0]}")/../..
 ALL_TARGETS=$(make -C "${KUBE_ROOT}" PRINT_HELP=y -rpn | sed -n -e '/^$/ { n ; /^[^ .#][^ ]*:/ { s/:.*$// ; p ; } ; }' | sort)
-CMD_TARGETS=$(ls -l "${KUBE_ROOT}/cmd" |awk '/^d/ {print $NF}')
+CMD_TARGETS=$(cd "${KUBE_ROOT}/cmd"; find . -mindepth 1 -maxdepth 1 -type d | cut -c 3-)
```
Comment (BenTheElder) [inline hack/make-rules/make-help.sh]: with gnu-find you get a warning if `-mindepth 1 -maxdepth1` (options) is not before `-type d` (find rules). updated

**Review by fejta — COMMENTED** [review]
/lgtm
/approve

**Comment by BenTheElder** [comment]
rebased

**Review by fejta — COMMENTED** [review]
/lgtm

**Pull request merged**